Integrations/integration-Panorama.yml

```diff
@@ -2299,6 +2324,396 @@ script:
+            'element': "<entry name='" + edl_name + "'></entry>",
+            'key': API_KEY
+        }
+        result = http_request(
+            URL,
+            'POST',
+            params = params,
+        )
+
+        return result
+
+
+    def panorama_delete_edl_command():
+        ''' Delete an EDL '''
+        edl_name = demisto.args()['name']
+
+        edl = panorama_delete_edl(edl_name)
+        edl_output = {'Name': edl_name}
+
+        demisto.results({
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['json'],
+            'Contents': edl,
+            'ReadableContentsFormat': formats['text'],
+            'HumanReadable': 'External Dynamic List was deleted successfully',
+            'EntryContext': {
+                  "Panorama.EDLs(val.Name == obj.Name)": edl_output
+            }
+        })
+
+
+    def panorama_refresh_edl(edl_name):
+        edl = panorama_get_edl(edl_name)
+        edl_type = ''.join(edl['type'].keys())
+
+        params = {
+            'type': 'op',
+            'cmd': '<request><system><external-list><refresh><type><' + edl_type + '><name>' + edl_name + '</name></' + edl_type + '></type></refresh></external-list></system></request>',
+            'key': API_KEY
+        }
+        result = http_request(
+            URL,
+            'POST',
+            params = params,
+        )
+
+        return result
+
+
+    def panorama_refresh_edl_command():
+        ''' Refresh an EDL '''
+        edl_name = demisto.args()['name']
+
+        result = panorama_refresh_edl(edl_name)
+
+        demisto.results({
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['json'],
+            'Contents': result,
+            'ReadableContentsFormat': formats['text'],
+            'HumanReadable': 'Refreshed External Dynamic List successfully',
+        })
+
+
+    @logger
+    def panorama_register_ip_tag(tag, ips, persistent = 1):
+        entry = ''
+        for ip in ips:
+            entry += '<entry ip=\"' + ip + '\" persistent=\"' + persistent + '\"><tag><member>' + tag + '</member></tag></entry>'
+
+        params = {
+            'type': 'user-id',
+            'cmd': '<uid-message><version>2.0</version><type>update</type><payload><register>' + entry + '</register></payload></uid-message>',
+            'key': API_KEY
+        }
+        result = http_request(
+            URL,
+            'POST',
+            params = params,
+        )
+
+        return result
+
+
+    def panorama_register_ip_tag_command():
+        ''' Register IPs to a Tag '''
+        tag = demisto.args()['tag']
+        ips = argToList(demisto.args()['IPs'])
+        persistent = demisto.args()['persistent'] if 'persistent' in demisto.args() else '1'
+
+        result = panorama_register_ip_tag(tag, ips, persistent)
+
+        registered_ip = {}
+        # update context only if IPs are persistent
+        if persistent == '1':
```

Comment:
shouldn't this be a boolean?

Reply:
no, it's a numeric value according to the api reference.